src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,592 @@
+    /**
+    Compute the probability that the current value plus anomaly score represents
+    an anomaly given the historical distribution of anomaly scores. The closer
+    the number is to 1, the higher the chance it is an anomaly.
+
+    @param value - the current metric ("raw") input value, eg. "orange", or
+                   '21.2' (deg. Celsius), ...
+    @param anomalyScore - the current anomaly score
+    @param timestamp - (optional) timestamp of the ocurrence,
+                       default (-1) results in using iteration step.
+    @return the anomalyLikelihood for this record.
+    **/
+    Real likelihood = 0.5f;
+
+    if (timestamp == -1) {
+      timestamp = this->iteration;
+    }
+
+    // store into relevant variables
+    this->runningRawAnomalyScores.push_back(anomalyScore); //TODO implement these vectors as sliding windows
+    auto newAvg = this->averagedAnomaly.compute(anomalyScore); 
+    this->runningAverageAnomalies.push_back(newAvg);
+    this->iteration++;
+    this->runningRawValues.push_back(rawValue); 
+    this->runningLikelihoods.push_back(likelihood);
+    
+    // We ignore the first probationaryPeriod data points - as we cannot reliably compute distribution statistics for estimating likelihood
+    if (this->iteration < this->probationaryPeriod) {
+      return 0.5f;
+    } //else {
+
+    auto anomalies = circularBufferToVector(this->runningAverageAnomalies); 
+    
+      // On a rolling basis we re-estimate the distribution
+      if ( this->iteration == 0 || (this->iteration % this->reestimationPeriod) == 0  || this->distribution.name == "unknown" ) {
+
+        auto numSkipRecords = this->calcSkipRecords_(this->iteration, this->runningAverageAnomalies.capacity(), this->learningPeriod); //FIXME this erase (numSkipRecords) is a problem when we use sliding window (as opposed to vector)! - should we skip only once on beginning, or on each call of this fn?
+        estimateAnomalyLikelihoods(anomalies, numSkipRecords);  // called to update this->distribution;  
+      }
+    
+    auto likelihoods = updateAnomalyLikelihoods(anomalies);
+
+      assert(likelihoods.size() > 0); 
+      likelihood = 1.0 - likelihoods[0]; 
+      assert(likelihood >= 0.0 && likelihood <= 1.0);
+
+    return likelihood;
+    }
+    
+private:
+    UInt learningPeriod = 288;
+    UInt reestimationPeriod = 100;
+  
+    DistributionParams distribution ={ "unknown", 0.0, 0.0, 0.0};   
+    UInt probationaryPeriod;
+    UInt iteration;
+    MovingAverage averagedAnomaly; // running average of anomaly scores
+    boost::circular_buffer<Real> runningLikelihoods; // sliding window of the likelihoods
+    boost::circular_buffer<Real> runningRawAnomalyScores; 
+    boost::circular_buffer<Real> runningAverageAnomalies; //sliding window of running averages of anomaly scores
+    boost::circular_buffer<Real> runningRawValues; // sliding window of the raw values
```

Comment:
TODO: actually, it seems only a few (runningAverageAnomalies) are used, should all other be deleted, or kept for "clarity"?